binding.c

```diff
@@ -192,7 +197,7 @@ on_udx_stream_read (udx_stream_t *stream, ssize_t read_len, const uv_buf_t *buf)
   if (n->mode == UDX_NAPI_FRAMED) {
     if (n->frame_len <= read) {
       n->frame_len = -1;
-    } else if (n->read_buf_free == 0) {
+    } else if (n->read_buf_free < 2 * UDX_MIN_BUF) {
       n->frame_len -= read;
     } else {
```

Comment:
shouldn't frame_len be decremented in the else here also?

Reply:
The idea is to keep reading until either:

1) `read` is equal to `frame_len`, after which the frame is complete and we reset it; or
2) `read_buf` is full, after which we decrease the `frame_len` by the amount read so far, flush the read buffer, and continue reading.

I _think_ the current logic is correct.

Reply:
but when is `frame_len` decremented if `read < frame_len` and `read_buf_free > UDX_MIN_BUF` ?

Reply:
It isn't as that indicates that the frame isn't complete yet, but there's still room for `read` to increase until the read buffer is flushed.

Reply:
I think I get it now